messaging/src/main/java/org/axonframework/eventhandling/pooled/Coordinator.java

```diff
@@ -591,6 +642,8 @@ public void run() {
                 return;
             }
 
+            executeUntilTrue(Coordinator.this::initializeTokenStore, 100L);
```

Comment:
Used an atomic boolean on the coordinator, so overhead is minimal.

Reply:
Looping in @Morlack for his opinion too.

Reply:
That is interesting, and I don't have a strong preference. I can easily move to start if @Morlack also agrees.

Reply:
> Are there reasonable scenarios where a users wants it's Event Processor(s) to not do anything, while the rest starts up as desired?
Seems unlikely to me, in my applications I would have wanted it to fail to boot. The only way it can fail is either misconfiguration or a downed database server. 

Note that the `TrackingEventProcessor` failing to initialize tokens would start up the application normally. Every worker loop the segments are checked on whether the store should be initialized.

Reply:
Which leads me to think, perhaps the initialization should take place in the `CoordinatorTask.claimNewSegments`, unless we want to diverge from the `TrackingEventProcessor` expectations

Reply:
we could have a private method to get the segments, witch initializes the tokens first when needed, and use that in the `CoordinatorTask.claimNewSegments` to get the segments?

Reply:
> Note that the TrackingEventProcessor failing to initialize tokens would start up the application normally. Every worker loop the segments are checked on whether the store should be initialized.

Although true, in an ideal world the number of TEP threads doesn't exceed the number of segments. 
The last segment would always be claimed by the `WorkerLauncher`, meaning the loop to initialize again isn't invoked anymore.

Granted, in a distributed environment, you'd always have TEP instances for the same `processorName` reapplying the check.

But, the PSEP will do this always, regardless, because the `Coordinator` will loop until it's told to shutdown.

Reply:
The thing that concerns me most is a potential performance degradation by moving this inside the `CoordinationTask`. Although it's not a functional change, it's not something users would expect.

Furthermore, referencing this:

> unless we want to diverge from the TrackingEventProcessor expectations

That is already the case in the current format, as we initialize the segments *before* we start the `CoordinationTask`.

Reply:
Alright, given these insights I would put the initialization in the lifecycle handler start method

Reply:
Put in the start method.